### PR TITLE
Tycho and Restructuring

### LINKED ISSUE #6: Implement a full command line build for BEF (Bazel rules or Tycho?)
We inherited the Bazel build rules for feature, plugin, and updatesite from the original Google project. When I split the feature into multiple plugins, it required changes to the rules. I put some effort into it, but it wasn't a high priority to fix them. Exporting the feature updatesite.zip from the Eclipse SDK is an ok solution at this early point in time.

The Bazel build is still used by CI for compiling the sources and running our unit/functional tests. But the actual packaging of the feature is left to Eclipse SDK. For iterative development this is fine because our team is able to just Run.. or Debug... the feature and the SDK auto packages the feature/plugins.

But in the long term, we should figure this out:
- invest in reviving the Bazel rules
- switch to Tycho for command line build

### PR DESCRIPTION
**Pull request by guw**
Fixes #6

### CODE REVIEW

**Comment by guw** [comment]
@plaird First draft of what to expect. I hope it's not too overwhelming. But moving to Tycho did require a restructuring.

**Review by plaird — APPROVED** [review]
@guw looks good, it will be nice to have a real command line build working again

File: docs/dev/thebuild.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
+## How the Build Works
```
Comment (guw) [inline docs/dev/thebuild.md:1]: @plaird Check out the new build doc. Let me know what I should add/document.

File: docs/dev/thebuild.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+- Choose *Configure Build Path* on *plugin-core*
+- Click on the *Java Build Path* item in the left nav
+- On the *Projects* tab, choose *Add...* and then pick the new project to depend on
+- Click on the *Project References* item in the left nav
+- Check the box next to the new item
+
+**Maven/Nexus Jars**
+Do not add free form jars to the Eclipse projects.
+Instead check if it is already available based on the Target Platform.
+If not you should check if its available in Eclipse Orbit.
```
Comment (plaird) [inline docs/dev/thebuild.md:55]: Nit: can you add a link to where to go for Orbit

File: docs/dev/thebuild.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+ - *Require-Bundle* for inter-project dependencies, i.e. dependencies to Eclipse plug-ins and to plug-ins in our project.
+ - *Import-Package* for third-party Maven/Nexus jars
+
+ Don't bother using version ranges.
+ They complicate things unnecessarily.
+ Use a lower version bound on *Import-Package*.
+
+ Because we do not use *Import-Package* between our plug-ins we do not set versions on exported packages.
+ If you believe we should, please raise an issue.
+ However, we do not expect anyone to provide alternate implementations of our packages.
```
Comment (plaird) [inline docs/dev/thebuild.md:72]: Agreed.

**Review by plaird — APPROVED** [review]

**Pull request merged**